internal/checker/terraform/rules/azure/verify_azure_instance_extensions.rego

```diff
@@ -0,0 +1,61 @@
+# METADATA
+# title: "Verify Virtual Machine Extensions are not Installed"
+# description: "This policy aims to enhance the security and management of virtual machines (VMs) by verifying that no additional or unnecessary virtual machine extensions are installed."
+# scope: package
+# related_resources:
+# - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine
+# custom:
+#   id: CB_TFAZR_022
+#   severity: MEDIUM
+package lib.terraform.CB_TFAZR_022
+
+isvalid(block) {
+	block.Type == "resource"
+	block.Labels[_] == "azurerm_linux_virtual_machine"
+}
+
+resource[resource] {
+	block := pass[_]
+	resource := concat(".", block.Labels)
+}
+
+resource[resource] {
+	block := fail[_]
+	resource := concat(".", block.Labels)
+}
+
+pass[block] {
+	block := input[_]
+	isvalid(block)
+	block.Labels[_] == "azurerm_linux_virtual_machine"
+	block.Attributes.allow_extension_operations == false
+}
+
+pass[block] {
```

Comment:
remove line 34 to 38 then create a supportedResources array with "azurerm_windows_virtual_machine"  to check isvalid in line 12